connectors/sources/confluence.py

```diff
@@ -1178,6 +1268,16 @@ async def _consumer(self):
             else:
                 yield item
 
+    def check_api_exceptions_and_raise(self):
+        failed_percentage = (
+            self.confluence_client.api_failed_count
+            / self.confluence_client.api_total_count
+        ) * 100
+        if failed_percentage >= 30:
```

Comment:
One day, I'd like to have logic similar to this at the framework level. But I have a few concerns about using error thresholds at the connector level like this.
1. first, this isn't currently configurable. IDK where you got the "30%" as a default, but I think you'd agree that no matter what default you pick, customers are going to want to change it. You didn't even do this as a constant that's easy to find and edit.
2. Even if it were configurable, it would need to be configurable specific to this connector. That means adding another RCF, and I don't think we want to keep polluting those.
3. Having this configurable for one connector makes all the other connectors look like they're missing the same feature.
4. Having this available is great, but it's only available for _some_ of the calls in this connector.

Reply:
> first, this isn't currently configurable. IDK where you got the "30%" as a default, but I think you'd agree that no matter what default you pick, customers are going to want to change it. You didn't even do this as a constant that's easy to find and edit.

Thank you for highlighting the lack of configurability and transparency in the default threshold setting. We acknowledge the need for flexibility in this aspect. If you believe it would be beneficial, we are open to discussing the possibility of removing the 30% logic to better align with user preferences.Please let us know your thoughts on this potential adjustment.

> Having this available is great, but it's only available for some of the calls in this connector.


The functionality mentioned is implemented for the api_call, it is also indirectly applicable to the paginated_api_call. This is because the paginated_api_call internally calls the api_call method to make API requests.

Reply:
Ok, after talking this over some with the team, I've changed my mind. We'll leave the threshold logic in this PR, and plan to revisit (potentially removing) when we're able to do this for the whole framework.

For now, can you pull the threshold default out as a constant, and let's set its value to 10%. For now, we won't expose as an RCF or via YAML config, but would require a code change in order to adjust the threshold.

Please add a release note section to this PR's description.

Reply:
Updated PR & Added release note, as you mentioned.